otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py

```diff
@@ -0,0 +1,331 @@
+
+        Args:
+            x1, y1, x2, y2 (int): Bounding box coordinates. # type: ignore
+            width (int): Width of image.
+            height (int): Height of image.
+
+        Returns:
+            List[int]: Generated bounding box.
+        """
+
+        def get_randomness(length: int) -> int:
+            if self.offset_bbox == 0:
+                return 0
+            return np.random.normal(0, min(length * 0.1, self.offset_bbox))
+
+        bbox = [
+            max(0, x1 + get_randomness(width)),
+            max(0, y1 + get_randomness(height)),
+            min(width, x2 + get_randomness(width)),
+            min(height, y2 + get_randomness(height)),
+        ]
+        return bbox
+
+    def generate_bbox_from_mask(self, gt_mask: np.ndarray, width: int, height: int) -> List[int]:
+        """Generate bounding box from given mask.
+
+        Args:
+            gt_mask (np.ndarry): Mask to generate bounding box.
+            width (int): Width of image.
+            height (int): Height of image.
+
+        Returns:
+            List[int]: Generated bounding box from given mask.
+        """
+        y_indices, x_indices = np.where(gt_mask == 1)
+        x_min, x_max = np.min(x_indices), np.max(x_indices)
+        y_min, y_max = np.min(y_indices), np.max(y_indices)
+        return self.generate_bbox(x_min, y_min, x_max, y_max, width, height)
+
+
+class OTXVisualPromptingDataModule(LightningDataModule):
+    """Visual Prompting DataModule.
+
+    Args:
+        config (Union[DictConfig, ListConfig]): Configuration.
+        dataset (DatasetEntity): Dataset entity.
+    """
+
+    def __init__(self, config: Union[DictConfig, ListConfig], dataset: DatasetEntity) -> None:
+        super().__init__()
+        self.config = config
+        self.dataset = dataset
+
+        self.train_otx_dataset: DatasetEntity
+        self.val_otx_dataset: DatasetEntity
+        self.test_otx_dataset: DatasetEntity
+        self.predict_otx_dataset: DatasetEntity
+
+    def setup(self, stage: Optional[str] = None) -> None:
+        """Setup Visual Prompting Data Module.
+
+        Args:
+            stage (Optional[str], optional): train/val/test stages, defaults to None.
+        """
+        if not stage == "predict":
+            self.summary()
+
+        image_size = self.config.image_size
+        if isinstance(image_size, int):
+            image_size = [image_size]
+
+        if stage == "fit" or stage is None:
+            train_otx_dataset = self.dataset.get_subset(Subset.TRAINING)
+            val_otx_dataset = self.dataset.get_subset(Subset.VALIDATION)
+
+            # TODO (sungchul): distinguish between train and val config here
+            train_transform = val_transform = MultipleInputsCompose(
+                [
+                    ResizeLongestSide(target_length=max(image_size)),
+                    Pad(),
+                    transforms.Normalize(mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375]),
```

Comment:
is this okay to hard coding the normalization factors instead of declaring in the configuration file?

Reply:
Those transforms will be used for train, validation, eval, export, etc.
So, I think it doesn't matter before a change of image encoder.

Additionally, I'm merging transform into dataset because the same transform will be used as I said.
Of course, it can be updated according to other implementations.
https://github.com/openvinotoolkit/training_extensions/blob/24b6b62cc5ce50eecee8c086becd24925f89e92d/otx/algorithms/visual_prompting/adapters/pytorch_lightning/datasets/dataset.py#L55-L59